### PR TITLE
EnBsb (Captain Keeta) OK

### PR DESCRIPTION
**Pull request by hensldm**
Thank you @Synray for the decomp work :) .

### CODE REVIEW

**Review by tom-overton — COMMENTED** [review]
Great work!

File: src/overlays/actors/ovl_En_Bsb/z_en_bsb.c [hunk 1]
```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bsb/func_80C0EEA0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bsb/func_80C0F078.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bsb/func_80C0F170.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bsb/EnBsb_Draw.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bsb/func_80C0F544.s")
+typedef enum {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bsb/z_en_bsb.c:207]:
```suggestion
typedef enum EnBsbAnimation {
```

File: src/overlays/actors/ovl_En_Bsb/z_en_bsb.c [hunk 2]
```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
...
+        Actor_PlaySfx(&this->actor, NA_SE_EN_KTIA_WALK);
+        Actor_RequestQuake(play, BREG(10) + 1, BREG(11) + 2);
+        if (Animation_OnFrame(&this->skelAnime, 4.0f) || Animation_OnFrame(&this->skelAnime, 8.0f)) {
+            func_80C0B31C(play, this, &this->unk_0304);
+        } else {
+            func_80C0B31C(play, this, &this->unk_02F8);
+        }
+    }
+
+    if (this->actor.xzDistToPlayer < BREG(7) + 400.0f) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bsb/z_en_bsb.c:466]:
```suggestion
    if (this->actor.xzDistToPlayer < (BREG(7) + 400.0f)) {
```

File: src/overlays/actors/ovl_En_Bsb/z_en_bsb.c [hunk 3]
```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
...
+            Rumble_Override(BREG(15), BREG(12) + sourceIntensity, BREG(13) + 15, BREG(14) + 70);
+        }
+    }
+}
+
+s32 func_80C0BC30(EnBsb* this) {
+    if ((this->actor.world.pos.z < -2180.0f) && (this->actor.world.pos.z > -2470.0f) &&
+        (fabsf(this->actor.world.pos.y - this->actor.home.pos.y) < 30.0f)) {
+        Vec3s sp38 = { 0x2EE0, -0x2710, 0xB5C8 };
+        Vec3s sp30 = { -0x2710, 0x2710, -0x2710 };
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bsb/z_en_bsb.c:485]:
```suggestion
        Vec3s sp38 = { 0x2EE0, -0x2710, -0x4A38 };
        Vec3s sp30 = { -0x2710, 0x2710, -0x2710 };
```
I think this makes more sense?

File: src/overlays/actors/ovl_En_Bsb/z_en_bsb.c [hunk 4]
```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
...
+        Flags_SetSwitch(play, this->switchFlag1);
+        this->unk_02A4 = 1;
+    }
+
+    if (this->unk_02A4 == 1) {
+        if ((play->gameplayFrames % 2) == 0) {
+            func_80C0B31C(play, this, &this->actor.world.pos);
+        }
+        Math_ApproachF(&this->actor.shape.shadowScale, 25.0f, 1.0f, 2.5f);
+        if (this->animEndFrame <= this->skelAnime.curFrame) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bsb/z_en_bsb.c:621]:
```suggestion
        if (this->skelAnime.curFrame >= this->animEndFrame) {
```

File: src/overlays/actors/ovl_En_Bsb/z_en_bsb.c [hunk 5]
```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
...
+
+    player->actor.world.rot.y = player->actor.shape.rot.y = this->actor.yawTowardsPlayer + 0x8000;
+
+    this->subCamEye.x = this->subCamEyeNext.x = Math_SinS(this->actor.yawTowardsPlayer) * 300.0f + this->unk_02E0.x;
+    this->subCamEye.y = this->subCamEyeNext.y = this->unk_02E0.y - 30.0f;
+    this->subCamEye.z = this->subCamEyeNext.z = Math_CosS(this->actor.yawTowardsPlayer) * 300.0f + this->unk_02E0.z;
+
+    this->subCamAt.x = this->subCamAtNext.x = Math_SinS(this->actor.yawTowardsPlayer) * 10.0f + this->unk_02E0.x;
+    this->subCamAt.y = this->subCamAtNext.y = this->unk_02E0.y - 10.0f;
+    this->subCamAt.z = this->subCamAtNext.z = Math_CosS(this->actor.yawTowardsPlayer) * 10.0f + this->unk_02E0.z;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bsb/z_en_bsb.c:1040]:
```suggestion
    this->subCamEye.x = this->subCamEyeNext.x = this->unk_02E0.x + Math_SinS(this->actor.yawTowardsPlayer) * 300.0f;
    this->subCamEye.y = this->subCamEyeNext.y = this->unk_02E0.y - 30.0f;
    this->subCamEye.z = this->subCamEyeNext.z = this->unk_02E0.z + Math_CosS(this->actor.yawTowardsPlayer) * 300.0f;

    this->subCamAt.x = this->subCamAtNext.x = this->unk_02E0.x + Math_SinS(this->actor.yawTowardsPlayer) * 10.0f;
    this->subCamAt.y = this->subCamAtNext.y = this->unk_02E0.y - 10.0f;
    this->subCamAt.z = this->subCamAtNext.z = this->unk_02E0.z + Math_CosS(this->actor.yawTowardsPlayer) * 10.0f;
```
I think this reads easier

File: src/overlays/actors/ovl_En_Bsb/z_en_bsb.c [hunk 6]
```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
...
+        this->subCamEyeNext.y = this->unk_02E0.y + 30.0f;
+        this->subCamEyeNext.z = cos + this->unk_02E0.z;
+
+        this->subCamAtNext.x = Math_SinS(this->actor.world.rot.y) * 10.0f + this->unk_02E0.x;
+        this->subCamAtNext.y = this->unk_02E0.y - 63.0f;
+        this->subCamAtNext.z = Math_CosS(this->actor.world.rot.y) * 10.0f + this->unk_02E0.z;
+    } else {
+        this->subCamAtNext.x = Math_SinS(this->actor.world.rot.y) * 10.0f + this->unk_02E0.x;
+        this->subCamAtNext.y = this->unk_02E0.y - 10.0f;
+        this->subCamAtNext.z = Math_CosS(this->actor.world.rot.y) * 10.0f + this->unk_02E0.z;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bsb/z_en_bsb.c:1087]:
```suggestion
        this->subCamEyeNext.x = this->unk_02E0.x + sin ;
        this->subCamEyeNext.y = this->unk_02E0.y + 30.0f;
        this->subCamEyeNext.z = this->unk_02E0.z + cos;

        this->subCamAtNext.x = this->unk_02E0.x + Math_SinS(this->actor.world.rot.y) * 10.0f;
        this->subCamAtNext.y = this->unk_02E0.y - 63.0f;
        this->subCamAtNext.z = this->unk_02E0.z + Math_CosS(this->actor.world.rot.y) * 10.0f;
    } else {
        this->subCamAtNext.x = this->unk_02E0.x + Math_SinS(this->actor.world.rot.y) * 10.0f;
        this->subCamAtNext.y = this->unk_02E0.y - 10.0f;
        this->subCamAtNext.z = this->unk_02E0.z + Math_CosS(this->actor.world.rot.y) * 10.0f;
```

File: src/overlays/actors/ovl_En_Bsb/z_en_bsb.c [hunk 7]
```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
...
+    this->subCamEye.y = this->subCamEyeNext.y = 375.0f;
+    this->subCamEye.z = this->subCamEyeNext.z = -1630.0f;
+
+    this->subCamAt.x = this->subCamAtNext.x = -360.0f;
+    this->subCamAt.y = this->subCamAtNext.y = 500.0f;
+    this->subCamAt.z = this->subCamAtNext.z = -2250.0f;
+
+    func_80C0BE1C(this, play);
+
+    switch (this->unk_1118) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bsb/z_en_bsb.c:1203]: Could you add `default: break;` to all 3 switch cases in this file?

File: src/overlays/actors/ovl_En_Bsb/z_en_bsb.c [hunk 8]
```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
...
+        }
+        if (this->collider.base.acFlags & AC_HIT) {
+            switch (this->actor.colChkInfo.damageEffect) {
+                case 14:
+                    if (this->unk_02DC) {
+                        var_s0 = 1;
+                        break;
+                    }
+
+                case 15:
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bsb/z_en_bsb.c:1440]:
```suggestion
                    }
                   // fallthrough
                case 15:
```

File: src/overlays/actors/ovl_En_Bsb/z_en_bsb.c [hunk 9]
```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
...
+                case 3:
+                    if (((this->drawDmgEffType != ACTOR_DRAW_DMGEFF_FROZEN_SFX) &&
+                         (this->drawDmgEffType != ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX)) ||
+                        (this->drawDmgEffTimer == 0)) {
+                        this->drawDmgEffTimer = 80;
+                        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FROZEN_SFX;
+                        var_s0 = -1;
+                        this->drawDmgEffScale = 0.0f;
+                        this->drawDmgEffFrozenSteamScale = 1.5f;
+                    }
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bsb/z_en_bsb.c:1480]:
```suggestion
                        this->drawDmgEffTimer = 80;
                        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FROZEN_SFX;
                        this->drawDmgEffScale = 0.0f;
                        this->drawDmgEffFrozenSteamScale = 1.5f;
                        var_s0 = -1;
                    }
```

File: src/overlays/actors/ovl_En_Bsb/z_en_bsb.c [hunk 10]
```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
...
+    this->actionFunc(this, play);
+    Actor_MoveWithGravity(&this->actor);
+
+    if (this->unk_02B0 == OBJECT_BSB_LIMB_NONE) {
+        func_80C0E618(this, play);
+        if ((!func_80C0E9CC(this, play)) && (this->unk_02B4 != 14)) {
+            this->actor.shape.rot.y = this->actor.world.rot.y;
+            this->actor.focus.pos.x = Math_SinS(this->actor.world.rot.y) * 20.0f + this->unk_02E0.x;
+            this->actor.focus.pos.y = this->unk_02E0.y;
+            this->actor.focus.pos.z = (Math_CosS(this->actor.world.rot.y) * 20.0f) + this->unk_02E0.z;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bsb/z_en_bsb.c:1611]:
```suggestion
            this->actor.focus.pos.x = this->unk_02E0.x + Math_SinS(this->actor.world.rot.y) * 20.0f;
            this->actor.focus.pos.y = this->unk_02E0.y;
            this->actor.focus.pos.z = this->unk_02E0.z + Math_CosS(this->actor.world.rot.y) * 20.0f;
```
note the extra set of brackets to be removed around `Math_CosS`

**Review by engineer124 — COMMENTED** [review]
Nice work! Just some minor nitpicks

**Review by engineer124 — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]
Nice :)

File: src/overlays/actors/ovl_En_Bsb/z_en_bsb.c [hunk 11]
```diff
@@ -124,123 +204,1689 @@ ActorInit En_Bsb_InitVars = {
...
+void func_80C0C6A8(EnBsb* this, PlayState* play) {
+    f32 curFrame = this->skelAnime.curFrame;
+
+    func_80C0B888(this, play);
+
+    if ((!this->unk_02DC || (this->unk_02DC && (this->animIndex == ENBSB_ANIM_2) && (curFrame >= this->animEndFrame) &&
+                             (this->unk_0294 == 0))) &&
+        ((this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_SFX) ||
+         (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX)) &&
+        (this->drawDmgEffTimer != 0)) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Bsb/z_en_bsb.c:745]: Gotta love those massive conditions

**Review by AngheloAlf — APPROVED** [review]

**Review by tom-overton — APPROVED** [review]

**Pull request merged**